On a related note, I found compiling fails for `DWIN_CREALITY_LCD` and `DWIN_LCD_PROUI` if `HAS_ZOFFSET_ITEM` is `1`:

<details><summary>details:</summary>
<p>

```prolog
Marlin/src/lcd/e3v2/creality/dwin.cpp: In function 'void hmiZoffset()':
Marlin/src/lcd/e3v2/creality/dwin.cpp:1387:36: error: 'PROBE_OFFSET_ZMIN' was not declared in this scope
 1387 |     LIMIT(hmiValues.offset_value, (PROBE_OFFSET_ZMIN) * 100, (PROBE_OFFSET_ZMAX) * 100);
      |                                    ^~~~~~~~~~~~~~~~~
Marlin/src/lcd/e3v2/creality/dwin.cpp:1387:63: error: 'PROBE_OFFSET_ZMAX' was not declared in this scope
 1387 |     LIMIT(hmiValues.offset_value, (PROBE_OFFSET_ZMIN) * 100, (PROBE_OFFSET_ZMAX) * 100);
      |                                                               ^~~~~~~~~~~~~~~~~
```

</p>
</details>

--- a/Marlin/src/lcd/e3v2/creality/dwin.cpp
+++ b/Marlin/src/lcd/e3v2/creality/dwin.cpp
@@ -1384,7 +1384,19 @@ void hmiMoveDone(const AxisEnum axis) {
       dwinUpdateLCD();
       return;
     }
-    LIMIT(hmiValues.offset_value, (PROBE_OFFSET_ZMIN) * 100, (PROBE_OFFSET_ZMAX) * 100);
+
+    #ifdef PROBE_OFFSET_ZMIN
+      #define _OFFSET_ZMIN (PROBE_OFFSET_ZMIN)
+    #else
+      #define _OFFSET_ZMIN -20
+    #endif
+    #ifdef PROBE_OFFSET_ZMAX
+      #define _OFFSET_ZMAX (PROBE_OFFSET_ZMAX)
+    #else
+      #define _OFFSET_ZMAX 20
+    #endif
+    LIMIT(hmiValues.offset_value, _OFFSET_ZMIN * 100, _OFFSET_ZMAX * 100);
+
     last_zoffset = dwin_zoffset;
     dwin_zoffset = hmiValues.offset_value / 100.0f;
     #if ANY(BABYSTEP_ZPROBE_OFFSET, JUST_BABYSTEP)

--- a/Marlin/src/lcd/e3v2/creality/dwin.cpp
+++ b/Marlin/src/lcd/e3v2/creality/dwin.cpp
@@ -1385,12 +1385,12 @@ void hmiMoveDone(const AxisEnum axis) {
       return;
     }
 
-    #ifdef PROBE_OFFSET_ZMIN
+    #if ENABLED(BABYSTEP_ZPROBE_OFFSET) && defined(PROBE_OFFSET_ZMIN)
       #define _OFFSET_ZMIN (PROBE_OFFSET_ZMIN)
     #else
       #define _OFFSET_ZMIN -20
     #endif
-    #ifdef PROBE_OFFSET_ZMAX
+    #if ENABLED(BABYSTEP_ZPROBE_OFFSET) && defined(PROBE_OFFSET_ZMAX)
       #define _OFFSET_ZMAX (PROBE_OFFSET_ZMAX)
     #else
       #define _OFFSET_ZMAX 20

--- a/Marlin/src/lcd/e3v2/proui/dwin.h
+++ b/Marlin/src/lcd/e3v2/proui/dwin.h
@@ -39,13 +39,6 @@
   #include "../../../feature/leds/leds.h"
 #endif
 
-#if ANY(BABYSTEPPING, HAS_BED_PROBE)
-  #define HAS_ZOFFSET_ITEM 1
-  #if !HAS_BED_PROBE
-    #define JUST_BABYSTEP 1
-  #endif
-#endif
-
 namespace GET_LANG(LCD_LANGUAGE) {
   #define _MSG_PREHEAT(N) \
     LSTR MSG_PREHEAT_##N                  = _UxGT("Preheat ") PREHEAT_## N ##_LABEL; \

